content/bestpractices/kubernetes_architecture_guide.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: bt_wiki
+title: Kubernetes Architecture Explained
+category: Manager Architecture
+draft: false
+abstract: Kubernetes Architecture Explained
+weight: 500
+aliases: /manager_architecture/create-deployment-flow/
+---
+
+# Kubernetes Architecture Explained
+
+## Cloudify Kubernetes support consists of three main components.
+
+* [Kubernetes Service Plugin](https://docs.cloudify.co/latest/working_with/official_plugins/orchestration/kubernetes/) - The Kubernetes plugin exposes the Kubernetes API into the Cloudify TOSCA based DSL. This will allow users to deploy a Kubernetes service by referencing directly your service template on your git repo or Helm chart. Cloudify [CI/CD integration](https://docs.cloudify.co/latest/working_with/integration/) also allows users to run Kubernetes services directly without having to create a Cloudify wrapper blueprint. In this case, the Ci/CD plugin takes care of dynamically binding the Cloudify blueprint behind the scene.
+* [Kubernetes Cluster Blueprint](https://github.com/cloudify-community/blueprint-examples/tree/master/kubernetes) - The Kubernetes blueprint provides a pre-canned template for provisioning a Kubernetes cluster. In the case of a managed Kubernetes such as GKE or EKS, the blueprint is responsible to set up the user environment. You can view such options [here](https://github.com/cloudify-community/blueprint-examples/tree/master/kubernetes/).
+* [Kubernetes Workflow](https://docs.cloudify.co/latest/working_with/official_plugins/orchestration/kubernetes/#workflows) - The workflow engine is responsible for handling install and uninstalls service workflow as well as day 2 operations such as updating of an existing service.
+
+![kubernetes_architecture]( /images/kubernetes_architecture/kubernetes_architecture.png )
+
+## Key features
+
+- Out of the box support for all major Kubernetes Platforms on private and public clouds (OpenShift, KubeSpray, GKE, EKS, and AKS).
+- Support multiple authentication methods Token- and “Kube config”- based authentication.
+- Resource state management - Block the deployment to a successful completion.
```

Comment:
@natishalom I don't understand this sentence

Reply:
It referes to the way Kuberenets works and how our plugin augment this.
The default behavior in Kubernetes is the deployment request is async means that the call returns before the resource have completed the deployment process.

We added behavior to block the deployment request till the resource is deployed successfully